pkg/segment/reader/segread/dechecker.go

```diff
@@ -38,13 +40,23 @@ returns:
 */
 func (sfr *SegmentFileReader) ApplySearchToMatchFilterDictCsg(match *structs.MatchFilter,
 	bsh *structs.BlockSearchHelper) (bool, error) {
+	var regexp *regexp.Regexp
+	var err error
 
 	if len(match.MatchWords) == 0 {
 		return false, nil
 	}
 
+	if match.MatchType == structs.MATCH_PHRASE && match.MatchOperator == utils.And {
```

Comment:
we can remove the `AND` operator check here (even though we had that check in the main method). So that in future we don't have to worry about changing this path. Basically we should pre-compile the regexp if the matchTYpe is match_phrase

Reply:
addressed in the latest commit.